[Doc] Revamp ray core design patterns doc [5/n]: ray get in submission order (#28345)
ray-project/ray: doc/source/ray-core/patterns/ray-get-submission-order.rst

Anti-pattern: Processing results in submission order using ray.get increases runtime
====================================================================================

**TLDR:** Avoid processing independent results in submission order using :ref:`ray.get() <ray-get-ref>` since results may be ready in a different order than the submission order.

> A batch of tasks are submitted, and we need to process their results individually once they’re done.
> If each task takes a different amount of time to finish and we process results in submission order, we may waste time waiting for all of the slower (straggler) tasks that are submitted earlier to finish while the other faster tasks have already finished.
> Instead, we want to process the tasks in the order that they finish using :ref:`ray.wait() <ray-wait-ref>` to speed up total time to completion.
> 

Reviewer: ```suggestion
A batch of tasks is submitted, and we need to process their results individually once they’re done.
If each task takes a different amount of time to finish and we process results in submission order, we may waste time waiting for all of the slower (straggler) tasks that were submitted earlier to finish while later faster tasks have already finished.

Instead, we want to process the tasks in the order that they finish using :ref:`ray.wait() <ray-wait-ref>` to speed up total time to completion.

```
